src/components/TodoList.tsx

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import { useContext } from 'react';
+import { MyContext } from './state';
+import { Filter } from '../Types/Filter';
+import React from 'react';
+import { Todo } from '../Types/Todo';
+
+export const TodoList: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    filter,
+    deleteTodo,
+    setEditingTodoId,
+    editingTodoId,
+  } = useContext(MyContext);
+
+  const filteredTodos = todos.filter((todo: Todo) => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.COMPLETED:
+        return todo.completed;
+      case Filter.ALL:
+        return true;
+      default:
+        return false;
+    }
+  });
+
+  function toggleTodoCompleted(id: number) {
+    setTodos((currentTodos: Todo[]) =>
+      currentTodos.map((todo: Todo) =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  }
+
+  function handleUpdateTodo(id: number, newTitle: string) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const handleUpdateTodo = (id: number, newTitle: string) => {
```